apps/tailwind-components/components/input/Ref.vue

```diff
@@ -0,0 +1,272 @@
+);
+
+onMounted(async () => {
+  tableMetadata.value = await fetchTableMetadata(
+    props.refSchemaId,
+    props.refTableId
+  );
+  //first we need to retrieve all selected items so are sure we have all for the template
+  if (
+    modelValue.value && Array.isArray(modelValue.value)
+      ? modelValue.value.length > 0
+      : modelValue.value
+  ) {
+    const data: ITableDataResponse = await fetchTableData(
+      props.refSchemaId,
+      props.refTableId,
+      { filter: { equals: modelValue.value } }
+    );
+    if (data.rows) {
+      hasNoResults.value = false;
+      data.rows?.forEach(
+        (row) => (selectionMap.value[applyTemplate(props.refLabel, row)] = row)
+      );
+    }
+  }
+
+  //then we load the options for the first time
+  await loadOptions({ limit: props.limit });
+  initialCount.value = count.value;
+});
+
+function applyTemplate(template: string, row: Record<string, any>): string {
+  const ids = Object.keys(row);
+  const vals = Object.values(row);
+  const label = new Function(...ids, "return `" + template + "`;")(...vals);
+  return label;
+}
+
+async function loadOptions(filter: IQueryMetaData) {
+  hasNoResults.value = true;
+  const data: ITableDataResponse = await fetchTableData(
+    props.refSchemaId,
+    props.refTableId,
+    filter
+  );
+
+  if (data.rows) {
+    hasNoResults.value = false;
+    data.rows.forEach(
+      (row) => (optionMap.value[applyTemplate(props.refLabel, row)] = row)
+    );
+    count.value = data.count;
+  } else {
+    hasNoResults.value = true;
+  }
+}
+
+function toggleSearch() {
+  showSearch.value = !showSearch.value;
+  if (!showSearch.value) updateSearch("");
+}
+
+function updateSearch(newSearchTerms: string) {
+  optionMap.value = {};
+  offset.value = 0;
+  searchTerms.value = newSearchTerms;
+  loadOptions({ limit: props.limit, searchTerms: searchTerms.value });
+}
+
+function select(label: string) {
+  console.log("select " + label);
+  if (!props.isArray) {
+    selectionMap.value = {};
+  }
+  selectionMap.value[label] = optionMap.value[label];
+  emit(
+    "update:modelValue",
+    props.isArray
+      ? Object.values(selectionMap.value).map((value) =>
+          extractPrimaryKey(value)
+        )
+      : extractPrimaryKey(optionMap.value[label])
+  );
+}
+
+function extractPrimaryKey(value: any) {
+  const result = {} as columnValueObject;
+  tableMetadata.value.columns
+    .filter((column) => column.key === 1)
+    .forEach((column) => {
+      result[column.id] = value[column.id];
+    });
+  return result;
+}
+
+function deselect(label: string) {
+  delete selectionMap.value[label];
+  emit(
+    "update:modelValue",
+    props.isArray
+      ? Object.values(selectionMap.value).map((value) =>
+          extractPrimaryKey(value)
+        )
+      : undefined
+  );
+}
+
+function clearSelection() {
+  selectionMap.value = {};
+  emit("update:modelValue", props.isArray ? [] : undefined);
+  updateSearch(""); //reset
+}
+
+function loadMore() {
```

Comment:
This causes content shifting or flickering. I would expect the load more to retrieve _n_ records after the current set. It should append them to the current list rather than rebuilding the list of options.

Reply:
Fix in another PR